### ISSUE TITLE
Create TestEnvironment 

### ISSUE THREAD

**Issue by jdegoes** [zio/zio#1129]
Inside the `testkit` project, we should stitch together a complete `TestEnvironment` to make it easy to test apps that just use ZIO modules; and to make it easy to use a wired-up `TestEnvironment` in other larger environments.

To complete this ticket, familiarity with `Runtime` / `DefaultRuntime` / `App`, and ZIO environment is required.

**Comment by LGLO** [comment]
In some interop projects we load all ZIO tests to have this enabled. This means scalajs tests from ZIO are compiled just to have base tests classes available.
I think testkit should be replacement for that pattern.

**Comment by dkarlinsky** [comment]
Hi, I'm really new to ZIO, but I am very interested in how it could make testing effectful code nicer.
I'm a big fan of leaning by doing, so I think contributing would be a great way to get into this.
I would like to take this issue, but I would likely need a bit more guidance. 
What would be the best channel for this?

**Comment by jdegoes** [comment]
@dkarlinsky Are you available to take this?

We need a version for JVM, something like this:

```scala
import zio.blocking._
  import zio.clock.Clock 
  import zio.console.Console
  import zio.scheduler.Scheduler
  import zio.random.Random
  import zio.system.System
  import zio.internal.Platform
  import zio.internal.PlatformLive   

object TestEnvironment {
  val Value: Managed[Nothing, DefaultRuntime#Environment] = 
    Managed.fromEffect {
      import zio.testkit._ 

      for {
        bootstrap <- ZIO.effect(PlatformLive.fromExecutionContext(scala.concurrent.ExecutionContext.Implicits.global))
        clockRef <- Ref.make(TestClock.DefaultData)
        consoleRef <- Ref.make(TestConsole.DefaultData)
        randomRef <- Ref.make(TestRandom.DefaultData)
        systemRef <- Ref.make(TestSystem.DefaultData)
      } yield new Blocking with Clock with Console with Random with Scheduler with System {
        val clock: TestClock = TestClock(clockRef)
        val console: TestConsole = TestConsole(consoleRef)
        val random: TestRandom = TestRandom(randomRef)
        val scheduler: TestScheduler = TestScheduler(clockRef, Runtime(clockRef, TestClock(clockRef)))
        val system: TestSystem = TestSystem(systemRef)
        val blocking: Blocking = new Blocking.Live {}
      }
    }
}
```

and one for `Scala.js`, without the `Blocking` (which does not exist for `Scala.js`). 

The JVM one should be placed in the JVM source directory, and the JS one, in the JS directory.

**Comment by adamgfraser** [comment]
I will take this.

**Comment by dkarlinsky** [comment]
Thanks @jdegoes, saw your answer just now...
Yes I would like to take it.
@adamgfraser do you mind if I gave it a shot?

**Comment by adamgfraser** [comment]
@dkarlinsky I am almost done with this one. Would you mind taking another ticket?

**Comment by dkarlinsky** [comment]
Ok, sure, @adamgfraser

**Comment by ghostdogpr** [comment]
Closed by #1255
